File: sic-api/Entities/Db/DbDistrict.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using sic_api.Entities;

namespace sic_api.Entities.Db;

[Index(nameof(DistrictCode), IsUnique = true)]
[Table("db_district")]
public class DbDistrict : BaseEntity
{
    
    [Required]
    [ForeignKey(nameof(Province))]
    [Column("province_id")]
    public Guid ProvinceId { get; set; }
    public DbProvince Province { get; set; } = default!;
    
    [Required]
    [MaxLength(10)]
    [Column("district_code")]
    public string DistrictCode { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("district_name_en")]
    public string DistrictNameEn { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("district_name_local")]
    public string DistrictNameLocal { get; set; } = default!;

    [Required]
    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    public ICollection<DbSubDistrict>? SubDistricts { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/db/DbDistrict.java
package com.softinter.sicapi.entity.db;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "db_district",
       indexes = {
           @Index(name = "idx_district_code", columnList = "district_code", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class DbDistrict extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "province_id", nullable = false)
    private DbProvince province;

    @Column(name = "district_code", nullable = false, length = 10)
    private String districtCode;

    @Column(name = "district_name_en", nullable = false, length = 255)
    private String districtNameEn;

    @Column(name = "district_name_local", nullable = false, length = 255)
    private String districtNameLocal;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @OneToMany(mappedBy = "district") // สมมติว่า DbSubDistrict มีฟิลด์ district
    private List<DbSubDistrict> subDistricts = new ArrayList<>();
}
